### PR TITLE
[FEATURE]: Authors frontend page generation

### PR DESCRIPTION
**Pull request by ihardz**
Static Badge

### 📃 Description

- @ihardz authors page generation.

----

### 🎟️ Issues

- #1879

----

### 📌 Type of change

> Please remove checbox options (below) that are not relevant to `this` Pull-Request.

- [ ] 📑 Chore update
> (^) Changes that invlove simple **chore** / **maintenance** changes, such as: documentation, folder/file simple restructure, syntax, typos, code clean, etc.

- [ ] 🐞 Bug fix
> (^) Changes involving **small** and **minor** **bug-fixes**, such as: fixing functionality, fixing faulty logic, fixing errors, fixing error handling/exceptions, etc.

- [x] ⚡️ New feature
> (^) Changes involving **minor** and **major** new **feature/enhancements**, such as: new widget, new improvements, new endpoints, new pages, new logic

- [ ] ❌ Breaking change
> (^) **Major** changes that will and/or can cause **issues and functionality to break**, such as: major project restructiring, major project feature change, major project bug fix, major project endpoints update, etc.

----

### 🧰 How Has This Been Tested?

> Please describe the tests that you ran to verify your changes,
> by simply checking the appropiate checkbox 

- [x] `localhost`
- [ ] `heroku (DEV)`

----

### ✔ Self-checklist

> Please select the checkboxes that apply below, as a developer **self-checklist**.

- [ ] This code follows the style guidelines of this project.
- [ ] This code is self-reviewed.
- [ ] This code is peer-reviewed.
- [ ] This code is properly commented, __particularly in hard-to-understand areas__.
- [ ] Documentation has been updated.
- [ ] This code does not generate new warnings.
- [ ] This code contains added tests that prove my fix is effective or that my feature works.
- [ ] This code new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

----

### ⚠ Warning(s)

- `console.(..)`

> (^) Please ensure that your code does not contain `console` without the `if (dev) console...`
> to avoid memory overflows on the server. Use the internal `debug.ts` project methods to gurantee no `poroduction`
> leaks of `console` errors. Or remove the `console` if no longer needed.

----

### CODE REVIEW

**Review by jonsnowpt — CHANGES_REQUESTED** [review]
@ihardz Deepsource issues that need review.

File: src/routes/(content)/a/[...permalink]/+page.svelte
Comment (migbash) [inline src/routes/(content)/a/[...permalink]/+page.svelte:1]: _this_ should be moved to another component called `Layout.svelte` iniside the `/src/components` directory.
Check the existing `src/(scores)/**/+page.svelte` example(s).

File: src/lib/components/content-section/AuthorArticleHeading.svelte
Comment (migbash) [inline src/lib/components/content-section/AuthorArticleHeading.svelte:1]: Unecessary component. Too simple and can be just pushed to the `parent` that is importing it (making use of).

File: src/lib/context/dark-theme-context.ts
Comment (migbash) [inline src/lib/context/dark-theme-context.ts:1]: Is there a need for this ?
We have exsiting methods that cover pretty much all aspects of JS dark mode handling from the `$userBetarenaSettings` store.

File: src/lib/components/shared/gradients/SideGradientWrap.svelte [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  export let dark: boolean = false;
+
+  export let side: "bottom" | "right" = "right"; // can be extended with "top" | "right" if will be needed
+  export let width: string = "100%";
+  export let height: string = "100%";
+  
+  export let hideOn: DeviceType[] = [];
+
+  const resolveAfterContent = (deviceType: DeviceType) =>
+    hideOn.includes(deviceType) ? "unset" : "\"\"";
```
Comment (migbash) [inline src/lib/components/shared/gradients/SideGradientWrap.svelte:13]: Typically, recommended to use the following for `listening` to device width:

```svelte
<svelte:window
  on:resize=
  {
    () => { return resizeCustom() }
  }
/>
```

Look for examples of this in the `components/` directory.

**Review by migbash — CHANGES_REQUESTED** [review]
#### [1] Make use of `?? (nullish)` instead of `|| (logical OR)` for fallback values.

#### [2] Update `<script> [..] </script>` to use a similar structure, that provided in the `.docs/.templates/*` files.

#### [3] Use `Allman` code structure for everything. 

Check the existing `src/components/**` files for example.

#### [+] check comments/requests added to files in _this_ `Pull-Request` for resolution.

File: src/lib/components/content-section/AuthorArticleContent.svelte [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    :global(h5),
+    :global(h6) {
+      font-weight: 800;
+      font-size: 20px;
+      margin: 40px 0 16px 0;
+    }
+
+    :global(h2) {
+      font-size: 22px;
+    }
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleContent.svelte:36]: I think you've miss understood, all I mean, is to remove the brackets `(..)` from `:global()`, nothing else. 

It should look like this:

```svelte
<style lang="scss">
  .author-article-content-component 
  {
    :global
    {
      a
      {
        /* 🎨 style */
        text-decoration: underline !important;
        display: inline;
      }

      h2,
      h3,
      h4,
      h5,
      h6
      {
        /* 🎨 style */
        font-weight: 800;
      }
    }
  }
</style>
```

File: src/lib/components/_main_/header/header-content/sub-components/BrandLogo.svelte [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<script lang="ts">
+  import betarenaLogoFull from '$lib/components/_main_/header/assets/betarena-logo-full.svg';
+  import betarenaLogoFullContrast from '$lib/components/_main_/header/assets/betarena-logo-full-contrast.svg';
+
+  import sessionStore from '$lib/store/session.js';
+	import { page } from '$app/stores';
+
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/_main_/header/header-content/sub-components/BrandLogo.svelte:8]: I don't think I understand your reasoining.

I am talking about the line :|: `export let dark: boolean = false;`.

We don't use `prop-drilling` in the project, such as you have done with the `{dark}` variable, which you pass from `+layout.svelte > HeaderContent.svelte > BrandLogo.svelte`, among other files in your pull-request.

Recommended to a `child` component to access such data via `context` and or `stores`, hence I suggested to use:

- getDarkThemeContext() or,
- $userBetarenaSettings.theme

, to obtain the `theme` state.

File: src/lib/components/content-section/AuthorArticleDetails/AuthorArticleDetails.svelte [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  import DetailsTemplateMobile from "./DetailsTemplateMobile.svelte";
+  import DetailsTemplateTablet from "./DetailsTemplateTablet.svelte";
+
+  let authorName = "Tiago Luchtenberg";
+  let authorHref = "/";
+  let date = "December 4, 2023";
+  let location = "Chicago, IL";
+  let readTime = "11 mins";
+  let duration = "2 days ago";
+  let ai = true;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleDetails/AuthorArticleDetails.svelte:11]: I think you have misunderstood, I am talking about `EsLint > (rule) one-var`, defined in the project `.eslintrc.yaml`:

https://eslint.org/docs/latest/rules/one-var#consecutive

Please enable the `EsLint`, as it appears it is not being refelected in your `Pull-Request`, which should give an error.

File: src/lib/components/content-section/AuthorArticleDetails/AuthorArticleDetails.svelte [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    .description {
+      font-weight: 400;
+      font-size: 12px;
+      line-height: 18px;
+    }
+
+    .tablet,
+    .desktop {
+      display: none;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleDetails/AuthorArticleDetails.svelte:58]: Change request stands. 

We do not use `display: none` in the project, and hide via `{if condition} {/if}`. 

Move from `CSS` breakpoints to `JS` breakpoints needs to be done.

1. `Blicking` does not happen, due to the existing `splashscreen` and with the `user-agent` decoding in the (server) 
`+layout.server.ts` that determines correct user device being used.

File: src/lib/components/content-section/AuthorArticleDetails/DetailsTemplateMobile.svelte [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        display: flex;
+        align-items: center;
+        line-height: 14px;
+      }
+      .time {
+        display: flex;
+        gap: 8px;
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleDetails/DetailsTemplateMobile.svelte:95]: Change stands, global styles need to be used defined for the `project`.

Please see `static/app.css` to see what is available , or use the `CTRL+SPACE` (MacOS) when typing inside a `class=""` to see available classes.

It's fully configured to follow the `Figma Guidelines` and it's used throughout the project, with no changes done that can cause a `break`.

Only custom `Styles/CSS` are to be placed inside the `<style> </style>` for a `.svelte` component.

File: src/lib/components/content-section/AuthorArticleDetails/AuthorArticleDetails.svelte [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    <DetailsTemplateTablet
+      {authorName}
+      {authorHref}
+      {date}
+      {location}
+      {readTime}
+      {duration}
+      {ai}
+    />
+  </div>
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleDetails/AuthorArticleDetails.svelte:36]: Change request stands.

Should be merged into `1 single` component, handled with `JS` conditionals and/or `CSS` to mutate a single component to adjust for repective `device` being used, in this case.

File: src/lib/components/content-section/AuthorArticleDetails/DetailsTemplateMobile.svelte [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  import IconPinLocation from "$lib/components/shared/icons/IconPinLocation.svelte";
+
+  export let avatar: string | undefined = undefined;
+  export let authorName: string;
+  export let authorHref: string;
+  export let date: string;
+  export let location: string;
+  export let readTime: string;
+  export let duration: string;
+  export let ai: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleDetails/DetailsTemplateMobile.svelte:13]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477946889

File: src/lib/components/_main_/auth/SignIn.svelte [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<script lang="ts">
+  import { page } from "$app/stores";
+  import { translationObject } from "$lib/utils/translation.js";
+  import ButtonHollow from "$lib/components/shared/buttons/ButtonHollow.svelte";
+  import AuthWidget from "$lib/components/_main_/auth/Auth_Widget.svelte";
+	import sessionStore from '$lib/store/session.js';
+
+  $: signInButtonLabel = $page.data.B_NAV_T?.scores_header_translations?.sign_in || translationObject?.sign_in;
+
+  const handleButtonClick = sessionStore.toggleAuthShow;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/_main_/auth/SignIn.svelte:10]: Perfectly readbale!

Change request stands for such cases where `JS` **one-off** logic/functions exist in a component.

File: src/lib/components/content-section/AuthorArticleContent.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleContent.svelte:36]: > have you tried this and it works?

Please try my last [comment](https://github.com/Betarena/scores/pull/1942#discussion_r1477925191) in your `local` environment, (most) components use this `structure` for `global { .. }` handle.

File: src/lib/components/_main_/header/header-content/HeaderContent.svelte [hunk 10]
```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+  import SignIn from "$lib/components/_main_/auth/SignIn.svelte";
+  import BrandLogo from "./sub-components/BrandLogo.svelte";
+  import ThemeToggle from "$lib/components/_main_/header/header-content/sub-components/ThemeToggle.svelte";
+
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/_main_/header/header-content/HeaderContent.svelte:6]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/_main_/header/header-content-mobile/HeaderContentMobile.svelte [hunk 11]
```diff
@@ -0,0 +1,34 @@
+<script lang="ts">
+  import ProfileAvatar from "$lib/components/shared/avatar/ProfileAvatar.svelte";
+  import ButtonSvgCircle from "$lib/components/shared/buttons/ButtonSvgCircle.svelte";
+  import BackButtonIconSrc from "$lib/components/shared/assets/chevron-left.svg";
+
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/_main_/header/header-content-mobile/HeaderContentMobile.svelte:6]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/_main_/header/header-content-mobile/HeaderContentMobile.svelte [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  let scrollTop;
+
+  $: buttonOpacity = scrollTop ? 0.8 : undefined;
+</script>
+
+<svelte:window bind:scrollY={scrollTop} />
+
+<header class="header-content-mobile-component">
+  <div class="wrap">
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/_main_/header/header-content-mobile/HeaderContentMobile.svelte:15]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477966597

File: src/lib/components/content-section/AuthorArticleDetails/DetailsTemplateTablet.svelte [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+  import IconPinLocation from "$lib/components/shared/icons/IconPinLocation.svelte";
+
+  export let avatar: string | undefined = undefined;
+  export let authorName: string;
+  export let authorHref: string;
+  export let date: string;
+  export let location: string;
+  export let readTime: string;
+  export let duration: string;
+  export let ai: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorArticleDetails/DetailsTemplateTablet.svelte:13]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477946889

File: src/lib/components/content-section/AuthorBadges.svelte [hunk 14]
```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+  import Tag from "$lib/components/shared/tags/Tag.svelte";
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/AuthorBadges.svelte:3]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/content-section/loaders/AuthorArticleContentLoader.svelte [hunk 15]
```diff
@@ -0,0 +1,21 @@
+<script lang="ts">
+  import LoaderInlineBlock from "$lib/components/shared/loaders/LoaderInlineBlock.svelte";
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/loaders/AuthorArticleContentLoader.svelte:3]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/content-section/loaders/AuthorArticleDetailsLoader.svelte [hunk 16]
```diff
@@ -0,0 +1,120 @@
+<script lang="ts">
+  import LoaderInlineBlock from "$lib/components/shared/loaders/LoaderInlineBlock.svelte";
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/loaders/AuthorArticleDetailsLoader.svelte:3]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/content-section/loaders/AuthorArticleImageLoader.svelte [hunk 17]
```diff
@@ -0,0 +1,15 @@
+<script lang="ts">
+  import LoaderInlineBlock from "$lib/components/shared/loaders/LoaderInlineBlock.svelte";
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/loaders/AuthorArticleImageLoader.svelte:3]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/content-section/loaders/AuthorBadgesLoader.svelte [hunk 18]
```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+  import LoaderInlineBlock from "$lib/components/shared/loaders/LoaderInlineBlock.svelte";
+  export let dark: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/content-section/loaders/AuthorBadgesLoader.svelte:3]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477941491

File: src/lib/components/shared/avatar/ProfileAvatar.svelte [hunk 19]
```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+  import ProfileAvatarNoneSrc from "$lib/components/shared/assets/profile-avatar-none.svg";
+  import { onMount, tick } from "svelte";
+  export let imageSrc: string | undefined = undefined;
+  export let imageAlt: string | undefined = undefined;
+  export let effect: boolean = false;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/shared/avatar/ProfileAvatar.svelte:6]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477946889

File: src/lib/components/shared/buttons/ButtonHollow.svelte [hunk 20]
```diff
@@ -0,0 +1,18 @@
+<script lang="ts">
+  export let label: string;
+  export let dark: boolean = true;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/shared/buttons/ButtonHollow.svelte:3]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477946889

File: src/lib/components/shared/buttons/ButtonSvgCircle.svelte [hunk 21]
```diff
@@ -0,0 +1,30 @@
+<script lang="ts">
+  export let src: string | undefined = undefined;
+  export let alt: string | undefined = undefined;
+  export let dark: boolean = true;
+  export let opacity: number | undefined = 1;
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/shared/buttons/ButtonSvgCircle.svelte:5]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477946889

File: src/lib/components/shared/gradients/SideGradientWrap.svelte [hunk 22]
```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+  import { DeviceType } from "$lib/context/device-type-context";
+
+  export let dark: boolean = false;
+
+  export let side: "bottom" | "right" = "right"; // can be extended with "top" | "right" if will be needed
+  export let width: string = "100%";
+  export let height: string = "100%";
+  
+  export let hideOn: DeviceType[] = [];
```
[the remark this answers is not in the document]
Comment (migbash) [reply src/lib/components/shared/gradients/SideGradientWrap.svelte:10]: see :|: https://github.com/Betarena/scores/pull/1942#discussion_r1477946889

**Pull request closed without merging**